arangod/VocBase/Methods/Queries.cpp

```diff
@@ -60,16 +63,71 @@ network::Headers buildHeaders() {
   return headers;
 }
 
-/// @brief return the list of currently running queries
-void getQueries(TRI_vocbase_t& vocbase, std::vector<aql::QueryEntryCopy> const& queries,
-                velocypack::Builder& out, char const* action, bool fanout) {
-  out.openArray();
+arangodb::Result checkAuthorization(TRI_vocbase_t& vocbase, bool allDatabases) {
+  Result res;
+
+  if (allDatabases) {
+    // list of queries requested for _all_ databases
+    if (!vocbase.isSystem()) {
+      // request must be made in the system database
+      res.reset(TRI_ERROR_ARANGO_USE_SYSTEM_DATABASE);
+    } else if (ExecContext::isAuthEnabled() &&
+               !ExecContext::current().isSuperuser()) {
+      // request must be made only by superusers (except authentication is turned off)
+      res.reset(TRI_ERROR_FORBIDDEN,
+                "only superusers are allowed to perform actions on all queries");
+    }
+  }
+
+  return res;
+}
+
+/// @brief return the list of currently running or slow queries
+arangodb::Result getQueries(TRI_vocbase_t& vocbase,
+                            velocypack::Builder& out, 
+                            arangodb::velocypack::StringRef action, 
+                            bool allDatabases, 
+                            bool fanout) {
+  Result res = checkAuthorization(vocbase, allDatabases);
+  
+  if (res.fail()) {
+    return res;
+  }
+
+  arangodb::DatabaseFeature& databaseFeature = vocbase.server().getFeature<DatabaseFeature>();
+
+  std::vector<arangodb::aql::QueryEntryCopy> queries;
   
   // local case
+  if (action == "slow") {
```

Comment:
Should we replace this with an enum? Or at least a `StaticString`?

Reply:
Ok, I have turned this into an enum.